app/src/main/java/com/victor/adasfeed/ProfileActivity.kt

```diff
@@ -47,20 +49,46 @@ class ProfileActivity : AppCompatActivity() {
 //            textUserName.text = getString(R.string.profile_name, safeUser.userName, safeUser.userNickname)
 
             user?.let { safeUser ->
-                textUserName.text = getString(R.string.profile_name, safeUser.userName)
-                textNickname.text = safeUser.userNickname
-                imageUser.setImageResource(safeUser.imageUser)
+                userIntent = safeUser
+                name = safeUser.userName
+                nickname = safeUser.userNickname
+                with(binding) {
+                    textUserName.text = safeUser.userName
+                    textNickname.text = safeUser.userNickname
+                    imageUser.setImageResource(safeUser.imageUser)
+                }
                 uri = safeUser.tel?.let {
                     Uri.parse("tel:${safeUser.tel}")
                 } ?: Uri.parse("tel:")
             }
         }
 
-        buttonCall.setOnClickListener {
-            val intent = Intent(Intent.ACTION_DIAL, uri)
-            startActivity(intent)
-            finish()
+        with(binding) {
+            buttonContact.setOnClickListener {
+                val intent = Intent(Intent.ACTION_DIAL, uri)
+                startActivity(intent)
+                finish()
+            }
+            editUserName.doAfterTextChanged {
+                name = it.toString()
+            }
+            editNickname.doAfterTextChanged {
+                nickname = it.toString()
+            }
+            saveButton.setOnClickListener {
+                textUserName.text = name
+                textNickname.text = nickname
+            }
+            buttonReturnToFeed.setOnClickListener {
+                val intent = Intent().apply {
+                    putExtra(EXTRA_KEY, userIntent.copy(userName = name, userNickname = nickname))
+                }
+                setResult(Activity.RESULT_OK, intent)
```

Comment:
Nice job bringing this content! 🚀 
But we normally use a standard navigation. We only use `setResult` when we open an Activity/Fragment specifically to GET some result than return, for example: opening camera intent: we want the picture of the result. In this case, we are just navigating to ProfileActivity and, if so, we change the information.

As our app is simple, we pass the information like this, but just to know the ideal scenario: we would call an API to save these datas, and we didn't have to pass this type of info updated. 😄